src/server.cpp

```diff
@@ -2509,8 +2509,13 @@ void Server::fillMediaCache()
 
 			const std::string &filename = dln.name;
 			if (m_media.find(filename) != m_media.end()) // Do not override
-				continue;
+			{
+				std::string parentpath = m_media.find(filename)->second.path;
 
+				// Check to make sure it isn't a seperate mod trying to override
+				if (parentpath.substr(0, parentpath.find("textures/")+9) == mediapath.substr(0, mediapath.find("textures/")+9))
```

Comment:
What happens if the texture is from a server texture pack? I guess they're not applied yet. Also, what about media folders etc? Might be better to add a field to the struct to store the mod name / source

Reply:
Server texturepacks override the original mod but mods that want to override the original mod will override the server texturepack from what I can tell